### PR TITLE
Implement DataProvider

### PR DESCRIPTION
**Pull request by chrismayer**
This adds a DataProvider component according to #28.

To resolve the dependency for ``geostyler-geojson-parser`` correctly this lets the package point to a special branch [with-dist-folder](https://github.com/terrestris/geostyler-geojson-parser/tree/with-dist-folder), which contains the compiled version of the GeoJSON parser. See https://github.com/terrestris/geostyler-geojson-parser/pull/5 for more information. 

The PR also contains a demo GUI to show the capabilities of the DataProvider.
**~~The demo UI is currently not runnable due to #32 (but can easily be fixed in local checkout).~~** 

image

### CODE REVIEW

**Comment by chrismayer** [comment]
The failing tests are related to the fact that the package ``geostyler-geojson-parser`` points to this [github-branch](https://github.com/terrestris/geostyler-geojson-parser/tree/with-dist-folder). While using ``npm link`` in the local setup everything is fine. Maybe we have to publish the ``geostyler-geojson-parser`` via npm to make this runnable in a sustainable way.

File: src/DataProvider/DataProvider.ts [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+  }
+
+  /**
+   * Imports the given data and transforms it into the internal data structure. Therefore an appropriate parser is
+   * detected by the given format.
+   * 
+   * @param inputData 
+   * @param format 
+   */
+  importData(inputData: any, format: string) {
```
Comment (KaiVolland) [inline src/DataProvider/DataProvider.ts]: I would switch the parameter order. Just a feeling.

**Comment by chrismayer** [comment]
I just added another commit, which includes the published packages for geostyler-data models
and the geostyler-geojson-parser from npmjs.org.

Since now Travis is green we might think about merging this PR.

**Comment by chrismayer** [comment]
Thnaks @KaiVolland!

**Pull request merged**